### PR TITLE
Pennylane 0.4 update and refactoring tests

### PR DESCRIPTION
**Pull request by quantshah**
- Updated expectations to observables and other changes such as pre_measure, obs_queue a/c to the latest Pennylane version

- Refactor of tests to pytests

### CODE REVIEW

**Comment by quantshah** [comment]
Strange pip error. @josh146

**Comment by josh146** [comment]
Ah, this is because occasionally Travis will run the tests inside a container with an old version of `pip`. To avoid this happening, you can add `pip install pip --upgrade` to the `.travis.yml` file, before installing any dependencies

**Comment by quantshah** [comment]
@josh146 Could you please take a look and approve this? We can merge this, make a new release and update the docs. Hopefully, this is enough for the qiskit plugin to work with the latest pennylane.

**Comment by josh146** [comment]
Will do that now!

File: pennylane_qiskit/devices.py [hunk 1]
```diff
@@ -27,7 +27,7 @@
     BasicAerQiskitDevice
     IbmQQiskitDevice
 
-See below for a description of the devices and the supported Operations and Expectations.
+See below for a description of the devices and the supported Operations and observables.
```
Comment (josh146) [inline pennylane_qiskit/devices.py]:
```suggestion
See below for a description of the devices and the supported operations and observables.
```

File: tests/test_device_initialization.py [hunk 2]
```diff
@@ -24,357 +24,565 @@
...
+
+    if args.device == "ibmq" or args.device == "all":
+        pytest.raises(
+            ValueError,
+            IbmQQiskitDevice,
+            wires=num_subsystems,
+            msg="Expected a ValueError if no IBMQX token is present.",
+        )
+
+    # put the IBMQX token back into place fo other tests to use
```
Comment (josh146) [inline tests/test_device_initialization.py]:
```suggestion
    # put the IBMQX token back into place for other tests to use
```

File: tests/test_device_initialization.py [hunk 3]
```diff
@@ -24,357 +24,565 @@
...
+
+num_subsystems = 4
+devices = None
+
+
+def test_ibm_no_token():
+    # if there is an IBMQX token, save it and unset it so that it doesn't interfere with this test
+    token_from_environment = os.getenv("IBMQX_TOKEN")
+    if token_from_environment is not None:
+        del os.environ["IBMQX_TOKEN"]
```
Comment (josh146) [inline tests/test_device_initialization.py]: How come you're deleting the IBMQX token environment variable here?

Comment (josh146) [reply tests/test_device_initialization.py]: Oh, I see, it's so that the next text will raise the correct error!

Comment (quantshah) [reply tests/test_device_initialization.py]: Yeah. Also, Carsten wrote those tests already.

**Pull request merged**